### PR TITLE
All claims 31 2 mg

### PR DESCRIPTION
**Pull request by markgreenburg**
## Description
Adds VA Medical Records supporting evidence type to the All Claims form.
- Allows addition of multiple facilities
- Allows association of disabilities to each VA evidence entry

## Testing done
- Tested locally
- Added unit tests around validations and required fields

## Screenshots
Empty form:
screen shot 2018-09-24 at 3 08 20 pm

-----
More empty form:
screen shot 2018-09-24 at 3 08 50 pm

-----
And some more:
screen shot 2018-09-24 at 3 09 06 pm

-----

State field goes away when country is not USA (per EVSS requirements):
screen shot 2018-09-24 at 3 09 29 pm

-----

Users can select multiple disabilities to associate with each piece of evidence:
screen shot 2018-09-24 at 3 09 36 pm

-----

Users can add multiple VA facilities on this page:
screen shot 2018-09-24 at 3 12 53 pm

## Acceptance criteria
- [x] Add VA facility evidence - one or many per mocks
- [x] Able to associate existing rated, new disabilities, or both with each VA facility

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/utils.jsx [hunk 1]
```diff
@@ -163,3 +163,49 @@ export function fetchPaymentInformation() {
...
+  const selectedRatedDisabilities = form.ratedDisabilities
+    .filter(disability => disability['view:selected']) || [];
+  const newDisabilities = form.newDisabilities || [];
+  const allSelectedDisabilities = selectedRatedDisabilities
+    .concat(newDisabilities);
+  const disabilitiesViews = allSelectedDisabilities.reduce((accum, curr) => {
+    const accumCopy = _.cloneDeep(accum);
+    const disabilityName = curr.name || curr.condition;
+    accumCopy[disabilityName] = { type: 'boolean' };
+    return accumCopy;
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/utils.jsx]: What do you think about using `set` instead of `cloneDeep`?

**Review by cvalarida — APPROVED** [review]
Just had a few comments and questions. The biggest one is whether we can uncomment that `depends` line.

File: src/applications/disability-benefits/all-claims/config/form.js [hunk 2]
```diff
@@ -181,6 +182,17 @@ const formConfig = {
         },
       }
     },
+    supportingEvidence: {
+      pages: {
+        vaMedicalRecords: {
+          title: 'VA Medical Records',
+          path: 'supporting-evidence/va-medical-records',
+          // depends: (formData) => formData['view:something'] === true,
```
[the remark this answers is not in the document]
Comment (markgreenburg) [reply src/applications/disability-benefits/all-claims/config/form.js]: This depends (🥁 ) on the 3.1 PR, which contains the actual data that this page will gate on.

**Review by peggygannon — CHANGES_REQUESTED** [review]
@markgreenburg Hi, Mark, please see below for a few content tweaks for this screen. 

the date question is a bit awkward, so let's rewrite it to: 
First date you received treatment for these conditions at this facility (this doesn't have to be exact).

Last date you received treatment for these conditions at this facility (this doesn't have to be exact).

Also, the conditions list from EVSS has all the condition names lowercased.. Is there something we can do make the first word of the condition capitalized when it's pulled in? Or would I need to go through the conditions list and edit and capitalize all first words?

**Comment by markgreenburg** [comment]
@peggygannon I actually already made the change to capitalize the conditions but neglected to update the screenshot for that section. Also, I've updated the wording on those date questions. Here's a new screenshot:

screen shot 2018-09-25 at 4 48 14 pm

**Review by peggygannon — APPROVED** [review]

File: src/applications/disability-benefits/all-claims/pages/vaMedicalRecords.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+          'ui:validations': [validateMilitaryTreatmentState],
+          'ui:options': {
+            expandUnder: 'country',
+            expandUnderCondition: USA
+          },
+          'ui:required': (formData, index) =>
+            _.get(
+              `vaTreatmentFacilities.${index}.treatmentCenterAddress.country`,
+              formData,
+              false
```
[the remark this answers is not in the document]
Comment (markgreenburg) [reply src/applications/disability-benefits/all-claims/pages/vaMedicalRecords.js]: Yes...yes we do. And besides, `false` isn't a great country choice 🤦‍♂️

**Pull request merged**